### PR TITLE
Extract rt types

### PR DESCRIPTION
**Pull request by ZenGround0**
First step in the effort to make runtime interface changes less painful to upgrade.  Extracts CBORer and LogLevel.  I put these in the network package since the network version is already important for runtime consumers.  Give me a better package / file name suggestion and I'll implement it.

### CODE REVIEW

File: network/runtime.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package network
+
+import "io"
+
+// These interfaces are intended to match those from whyrusleeping/cbor-gen, such that code generated from that
+// system is automatically usable here (but not mandatory).
+type CBORMarshaler interface {
```
Comment (Stebalien) [inline network/runtime.go:7]: I'd actually just put these into a separate `cbor` package.

File: network/runtime.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	UnmarshalCBOR(r io.Reader) error
+}
+
+type CBORer interface {
+	CBORMarshaler
+	CBORUnmarshaler
+}
+
+// Specifies importance of message, LogLevel numbering is consistent with the uber-go/zap package.
+type LogLevel int
```
Comment (Stebalien) [inline network/runtime.go:21]: Maybe a new `log` package? Maybe just introduce a runtime package?  To reduce imports, we can type alias any of these inside specs-actors anyways.

File: network/runtime.go (hunk 1 above)
Comment (anorth) [reply network/runtime.go:7]: SG. `cbor.Marshaler`, `cbor.Er` etc :)

File: network/runtime.go (hunk 2 above)
Comment (anorth) [reply network/runtime.go:21]: `log` could be an annoying collision for consumers. `runtime` or `rt` could be ok, tho I would then wish that `ExitCode` was there too.

File: network/runtime.go (hunk 1 above)
Comment (anorth) [reply network/runtime.go:7]: or `ser`, `serde`?

**Review by anorth — APPROVED** [review]

File: network/runtime.go (hunk 2 above)
Comment (ZenGround0) [reply network/runtime.go]: Going with `rt`.  Will file an issue for moving exitcodes there as a nice to have if things are going really well early next week.

File: network/runtime.go (hunk 1 above)
Comment (ZenGround0) [reply network/runtime.go]: Going with (Un)Marshaler and Er

**Pull request merged**